tests/util/test_networks.py

```diff
@@ -192,3 +259,44 @@ def test_input_validation_on_ema_norm():
         networks.EMANorm(128, decay=-0.1)
 
     networks.EMANorm(128, decay=0.05)
+
+
+def run_ema_norm(norm, random_tensor):
+    moving_random_tensor = random_tensor.clone()
+    for _ in range(100):
+        # calculating EMA of a moving distribution
+        # moving_random_tensor += 1
+        norm(moving_random_tensor)
+
+
+@pytest.mark.parametrize("decay_within_batch", [False, True])
+@pytest.mark.parametrize("decay", [0.5, 0.99])
+def test_ema_norm_batch_correctness(decay_within_batch, decay):
+    norm_for_incremental = EMANormIncremental(
+        num_features=256,
+        decay=decay,
+        decay_within_batch=decay_within_batch,
+    )
+    norm_for_batch = networks.EMANorm(256, decay, decay_within_batch=decay_within_batch)
+    random_tensor = th.randn(64, 256)
```

Comment:
Consider parameterizing the shape? In particular might be useful to try a tensor that has length 1 (just batch dimension), as you've got some logic to special-case that in `EMANorm`, would be good to check it works.